hles/transformation/src/main/scala/org/broadinstitute/monster/dap/environment/TemperaturePrecipitationTransformations.scala

```diff
@@ -0,0 +1,116 @@
+package org.broadinstitute.monster.dap.environment
+
+import org.broadinstitute.monster.dap.RawRecord
+import org.broadinstitute.monster.dogaging.jadeschema.fragment.EnvironmentTemperaturePrecipitation
+
+object TemperaturePrecipitationTransformations {
+
+  /**
+    * Parse all temperature and precipitation variables out of a raw RedCap record,
+    * injecting them into a partially-modeled environment record.
+    */
+  def mapTemperaturePrecipitationVariables(
+    rawRecord: RawRecord
+  ): EnvironmentTemperaturePrecipitation = {
+    EnvironmentTemperaturePrecipitation(
+      tpDataYear = rawRecord.getOptionalNumber("tp_data_year"),
+      tpPcpnAnnual01 = rawRecord.getOptionalNumber("tp_pcpn_annual_01").map(_.toDouble),
```

Comment:
it would be so great if we didn't have to write all this out since it is so repeateddd ugh :/ I don't think we have a good way to do so thus far